lib/AST/ASTWalker.cpp

```diff
@@ -146,6 +146,19 @@ class Traversal : public ASTVisitor<Traversal, Expr*, Stmt*,
     for (Decl *M : ED->getMembers()) {
       if (doIt(M))
         return true;
+
+      if (Walker.shouldWalkAccessorsTheOldWay()) {
+        // Pretend that accessors share a parent with the storage.
+        //
+        // FIXME: Update existing ASTWalkers to deal with accessors appearing as
+        // children of the storage instead.
+        if (auto *ASD = dyn_cast<AbstractStorageDecl>(M)) {
+          for (auto AD : ASD->getAllAccessors()) {
+            if (doIt(AD))
+              return true;
+          }
+        }
+      }
```

Comment:
Consider extracting a helper function for this so that it doesn't have to be repeated in every DC.

Reply:
This is a compatibility behavior that's going away soon.